src/player.h

```diff
@@ -1461,6 +1461,12 @@ class player : public Character, public JsonSerializer, public JsonDeserializer
         std::set<tripoint> get_path_avoid() const override;
 
         /**
+         * Makes the NPC considered to be attacked by the player first if not already hostile.
+         * Angers the NPC to allow for fight or flight.
+         */
+        void aggress_npc( npc &target );
```

Comment:
I can do something like `npc::make_assaulted_from( const player & u )` but at this time the player ref is not needed in any of the actual functionality. If the NPCs tracked opinion by player / creature instances it would be useful, but right now that isn't the case. I'm okay with leaving the signature like that if there is a way we can get around the build issue of an unused parameter.